tagstudio/src/core/library.py

```diff
@@ -894,19 +894,30 @@ def refresh_dir(self) -> Generator:
                     and "tagstudio_thumbs" not in f.parts
                     and not f.is_dir()
                 ):
-                    if f.suffix.lower() not in self.ext_list and self.is_exclude_list:
+                    suffix = (
+                        f.name.lower() if f.name.startswith(".") else f.suffix.lower()
+                    )
+                    if suffix not in self.ext_list and self.is_exclude_list:
                         self.dir_file_count += 1
                         file = f.relative_to(self.library_dir)
                         if file not in self.filename_to_entry_id_map:
                             self.files_not_in_library.append(file)
-                    elif f.suffix.lower() in self.ext_list and not self.is_exclude_list:
+                    elif suffix in self.ext_list and not self.is_exclude_list:
                         self.dir_file_count += 1
                         file = f.relative_to(self.library_dir)
                         try:
                             _ = self.filename_to_entry_id_map[file]
                         except KeyError:
                             # print(file)
                             self.files_not_in_library.append(file)
+                    else:
+                        entry_id = self.filename_to_entry_id_map.get(
+                            f.relative_to(self.library_dir), None
+                        )
+
+                        if entry_id:
+                            # A exclude file is in the library, let's remove it.
+                            self.remove_entry(entry_id)
```

Comment:
Not touching file entries that fall under the exclusion list is intended behavior. This would not only wipe entry data when playing with the exclusion list, but the user wouldn't even be notified of it.

Reply:
I will remove this then!